packages/trpc/server/routers/viewer/apps.tsx

```diff
@@ -74,34 +62,35 @@ export const appsRouter = router({
             dirName: app.dirName || app.slug,
             enabled: dbData?.enabled || false,
             isTemplate: app.isTemplate,
-          });
-        } else {
-          const keysSchema = appKeysSchemas[app.dirName as keyof typeof appKeysSchemas];
+          };
+        }
 
-          const keys: Record<string, string> = {};
+        const keysSchema = appKeysSchemas[app.dirName as keyof typeof appKeysSchemas];
 
-          if (typeof keysSchema !== "undefined") {
-            Object.values(keysSchema.keyof()._def.values).reduce((keysObject, key) => {
-              keys[key as string] = "";
-              return keysObject;
-            }, {} as Record<string, string>);
-          }
+        const keys: Record<string, string> = {};
 
-          filteredApps.push({
-            name: app.name,
-            slug: app.slug,
-            logo: app.logo,
-            type: app.type,
-            title: app.title,
-            description: app.description,
-            enabled: dbData?.enabled || false,
-            dirName: app.dirName || app.slug,
-            keys: Object.keys(keys).length === 0 ? null : keys,
-          });
+        // `typeof val === 'undefined'` is always slower than !== undefined comparison
+        // it is important to avoid string to string comparisons as much as we can
+        if (keysSchema !== undefined) {
```

Comment:
I think this is causing a type error:

<img width="732" alt="image" src="https://user-images.githubusercontent.com/3504472/232637119-e9d5e5cb-effc-4db7-a921-8df768c24c41.png">

Reply:
Pushed a fix